app/admin/allocation_chart.rb

```diff
@@ -11,9 +13,33 @@
           column(I18n.t('name'), :user)
           column(I18n.t('skills')) { |allocation| allocation.user.skills.map(&:title).join(', ') }
           column(I18n.t('client'), :project)
-          column(I18n.t('allocated_until'), :end_at)
+          column(I18n.t('allocated_until'), class: 'allocated-column') do |allocation|
+            build_allocation_status_cell(allocation)
+          end
         end
       end
     end
   end
 end
+
+def build_allocation_status_cell(last_allocation)
+  allocation_status = ActiveAdmin::AllocationChartHelper.allocation_status_for(last_allocation: last_allocation)
+
+  unless last_allocation.end_at.nil?
+    return column_cell_base(content: last_allocation.end_at, allocation_status: allocation_status)
+  end
+
+  content = if allocation_status == ActiveAdmin::AllocationChartHelper::Status::NOT_ALLOCATED
```

Comment:
for the sake of making this method smaller, is it possible to move this if/else into a separate method? then you'd call it in here, like so:
```ruby
content = allocation_status_content
```

you could even pass the method directly as an argument in the `content` key in line 38:
```ruby
column_cell_base(content: allocation_status_content, allocation_status: allocation_status)
```
(the name is just a suggestion, feel free to name something more appropriate)